### ISSUE TITLE
Touchable components stop responding randomly

### ISSUE THREAD

**Issue by vrgimael** [facebook/react-native#36710]
### Description

We've been experiencing this issue in a couple of our apps, it's kinda driving us mad.

Basically, sometimes, completely unpredictably the touchable areas just stop responding.

We tried reproducing this consistently in several different ways but just couldn't pin point what causes it to happen and in which devices. It happens on both iOS and Android, and also in debug or release mode. Generally, it looks like it only happens after the user has been using the app for a little while (10+ minutes). It seems to happen in any button in the app randomly, and some times it also seemingly randomly starts working again. Closing and opening the app again always fixes the problem.

We had clients complain about it in quite a few occasion, but some of our public testers rarely ever experience this either which makes it even harder to understand.

I spent a very long time investigating it over literally the past entire year, but it's really really hard to debug, because when the issue happens in debug mode, anything you do in the code that causes hot reloading actually "fixes it", making it almost impossible to investigate.

I have added lots of debugging tools to try and understand what happens though, I got a few bits of information that might be able to help.

1. This is not a performance or execution queue issue, as we have other interactive components in the screen that update automatically based on time and these never freeze.
2. Other touch based components seem to continue working like scrolls or maps.
3. The zindex of the buttons I'm trying to interact with is much greater than anything else in the screen.
4. I have tried swapping out our "button" component's touchable with react-native-gesture-handler's, but the issue still persists.
5. This has been going for several different versions of RN and none resolved.
6. When the issue happens, the onPressIn and onPressOut events are fired almost simultaneously with only a few milliseconds difference but onPress is not fired - kinda like what happens when you're scrolling or actually interacting with another gesture handler. I tried to make a fallback mechanism that uses this behaviour to our advantage but it's outcome is not very consistent for a few different reasons.
7. Surprisingly (and this is what makes me think it's an issue with RN and not our code), when this issue happens and you manage to open the RN inspect menu by shaking the device, the tabs in the menu (Inspect, Perf, Network, Touchables) don't even work, exactly the same behaviour observed with our own buttons in the app.
8. It might be out of pure coincidence, but I have only seen this happen in apps that use react-native-maps - one of my hypothesis is that the gesture responder somehow gets confused with the map handler.

As far as I can tell, this has always happened - it's so rare that I was not very bothered by it. But as our apps grow and have a larger userbase now, this issue is coming up more frequently and we need to figure it out some time soon.
I suspect this is not too much of a problem in other apps because normally if a problem like this happens you'd just reopen the app and everything is fine again. However, it has already caused bad reviews in the App Store because our app's functionality is really time sensitive and if users aren't able to interact with the screen it's kinda problematic.

I've read all issues related to several relevant keywords in this repo (like touchable, onpressin, not responding, etc...), and there are a few issues that mention similar behaviour https://github.com/facebook/react-native/issues/34999 https://github.com/facebook/react-native/issues/36063 but none have barely any responses (possibly due to the rare nature of the problem).

Any help would be really appreciated. Please let me know if I can provide any other useful information.

Thanks

### React Native Version

0.70.6

### Output of `npx react-native info`

System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 192.88 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.19.1 - ~/.yarn/bin/yarn
    npm: 7.24.2 - ~/CodeLeap-Software/Client-Projects/x-follo-mobile/mobile/node_modules/.bin/npm
    Watchman: 2022.07.04.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 26, 28, 29, 30, 31, 33
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
      System Images: android-30 | Google Play ARM 64 v8a, android-31 | ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-31 | Google Play ARM 64 v8a, android-Sv2 | Google APIs ARM 64 v8a, android-Sv2 | Google Play ARM 64 v8a
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.13 - /Users/victor/Library/Java/JavaVirtualMachines/azul-11.0.13/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^18.2.0 => 18.2.0 
    react-native: ^0.70.6 => 0.70.7 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

The hardest thing about this is that it can't be reliably reproduced. See description above.

### Snack, code example, screenshot, or link to a repository

Example in our own app (that's in the App Stores too), a client sent us this recording of it happening in practice. Looking back now I should have taken more recordings of it but I'm not sure it would be of any use too.

https://user-images.githubusercontent.com/14969223/228675821-fb9b1eac-7228-4276-85ce-5d1c2bf2f284.mp4

EDIT 30/05/23 -> Here are the dependencies for this project:
```
  "dependencies": {
    "@babel/plugin-proposal-export-namespace-from": "^7.18.9",
    "@babel/runtime": "^7.14.6",
    "@faker-js/faker": "^7.3.0",
    "@googlemaps/polyline-codec": "^1.0.1",
    "@invertase/react-native-apple-authentication": "^2.2.2",
    "@miblanchard/react-native-slider": "^2.3.1",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-native-community/netinfo": "^9.3.6",
    "@react-native-community/push-notification-ios": "^1.10.1",
    "@react-native-firebase/analytics": "^15.2.0",
    "@react-native-firebase/app": "^15.2.0",
    "@react-native-firebase/auth": "^15.2.0",
    "@react-native-firebase/crashlytics": "^15.2.0",
    "@react-native-firebase/messaging": "^15.2.0",
    "@react-native-google-signin/google-signin": "^8.0.0",
    "@react-native-masked-view/masked-view": "^0.2.8",
    "@react-navigation/bottom-tabs": "^6.3.2",
    "@react-navigation/native": "^6.0.11",
    "@react-navigation/stack": "^6.2.2",
    "@sentry/react-native": "^3.1.1",
    "callsites": "^3.1.0",
    "date-fns": "^2.29.2",
    "deep-equal": "^2.0.5",
    "deep-object-diff": "^1.1.9",
    "deprecated-react-native-prop-types": "^2.3.0",
    "expo": "^48.0.6",
    "expo-av": "^13.0.2",
    "expo-battery": "^7.0.0",
    "expo-cellular": "^5.0.1",
    "expo-constants": "^14.0.2",
    "expo-keep-awake": "^11.0.1",
    "expo-linear-gradient": "^12.1.2",
    "expo-linking": "^3.2.3",
    "expo-local-authentication": "^13.0.2",
    "expo-location": "^15.0.1",
    "expo-network": "^5.0.0",
    "expo-sensors": "^12.0.1",
    "expo-store-review": "^6.0.0",
    "expo-task-manager": "^11.0.1",
    "faker": "^5.5.3",
    "fs": "0.0.1-security",
    "geolocation-utils": "^1.2.5",
    "libphonenumber-js": "^1.9.25",
    "lodash": "^4.17.21",
    "lottie-ios": "3.4.1",
    "lottie-react-native": "5.1.3",
    "moment": "^2.29.1",
    "npm": "^7.18.1",
    "npm-license-crawler": "^0.2.1",
    "patch-package": "^6.4.7",
    "path": "^0.12.7",
    "pretty-bytes": "^5.6.0",
    "react": "^18.2.0",
    "react-mixin": "^5.0.0",
    "react-native": "^0.70.6",
    "react-native-android-open-settings": "^1.3.0",
    "react-native-animatable": "^1.3.3",
    "react-native-background-timer": "^2.4.1",
    "react-native-contacts": "^7.0.4",
    "react-native-date-picker": "^4.2.5",
    "react-native-device-info": "^10.3.0",
    "react-native-fbsdk-next": "^10.0.0",
    "react-native-flipper": "0.163.0",
    "react-native-geocoding": "^0.5.0",
    "react-native-geolocation-service": "^5.3.0-beta.1",
    "react-native-gesture-handler": "^2.8.0",
    "react-native-get-random-values": "^1.8.0",
    "react-native-google-mobile-ads": "^9.1.1",
    "react-native-google-places-autocomplete": "^2.5.1",
    "react-native-gradle-plugin": "^0.70.0",
    "react-native-iap": "^12.7.2",
    "react-native-image-crop-picker": "^0.38.0",
    "react-native-in-app-review": "^4.3.1",
    "react-native-in-app-utils": "^6.1.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-localize": "^2.2.2",
    "react-native-maps": "^1.3.2",
    "react-native-notification-sounds": "^0.5.5",
    "react-native-onesignal": "^4.5.1",
    "react-native-permissions": "^3.6.0",
    "react-native-pose": "^0.9.1",
    "react-native-push-notification": "^8.1.1",
    "react-native-render-html": "^6.3.4",
    "react-native-ringer-mode": "^2.0.0",
    "react-native-safe-area-context": "^4.3.1",
    "react-native-screens": "^3.15.0",
    "react-native-splash-screen": "^3.3.0",
    "react-native-svg": "^13.4.0",
    "react-native-system-navigation-bar": "^1.0.5",
    "react-native-system-setting": "^1.7.6",
    "react-native-vector-icons": "^9.2.0",
    "react-redux": "^8.0.2",
    "react-timer-hook": "^3.0.5",
    "react-timer-mixin": "^0.13.4",
    "react-use": "^17.4.0",
    "redux": "^4.2.0",
    "redux-flipper": "^2.0.2",
    "redux-middleware-flipper": "^1.0.9",
    "redux-sentry-middleware": "^0.2.2",
    "redux-thunk": "^2.4.1",
    "rn-async-storage-flipper": "^0.0.10",
    "rn-emoji-keyboard": "^1.0.0",
    "rn-sliding-up-panel": "^2.4.6",
    "url-parse": "^1.5.10",
    "uuid": "^8.3.2",
    "validator": "^13.7.0"
  },
```

**Comment by necolas** [comment]
I think it will be difficult to get attention from the maintainers without a) providing a test case, b) steps to reproduce the issue in the test case, and c) verifying the issue is still present after upgrading to the latest version of React Native

**Comment by vrgimael** [comment]
Hi @necolas thanks for the feedback, however you really should not close this issue..... other users have mentioned similar problems, please do not ignore this issue just because it happens intermittently. 

a) As I said, the problem cannot be reliably reproduced at all
b) The problem happens intermittently
c) This has been happening through several previous versions of RN... I will upgrade again but this certainly is not the problem.

I provided as much information as I could possibly gather from this and hoped it could be helpful to others too.

**Comment by Slapbox** [comment]
We may be experiencing this issue as well with any version above 0.70.5 - only it's entirely predictable here and prevents signing into our app, and therefore is a total showstopper for us. 0.70.8 and 0.70.6 both exhibit this behavior but 0.70.5 does not. It's possible we're seeing a different issue though.

I have no idea how I would create a repro for this because it's such a basic issue, just a button, surely anything easily reproduced would have been caught in development or testing by the React Native team.

**Comment by israelidanny** [comment]
For what its' worth, I'm also experiencing it, and it's very hard to understand how to give a reproducible example - because it doesn't happen reliably. But in an intermittent fashion onPress just won't fire for a bunch of times in a row.

**Comment by vrgimael** [comment]
Hmm I am not using the new architecture though, and this problem seems to have happened for a long time (before new architecture existed).

We tried this example on our project and it doesn't reproduce the issue unfortunately...

Any thoughts would be appreciated.

**Comment by NiuGuohui** [comment]
> Hmm I am not using the new architecture though, and this problem seems to have happened for a long time (before new architecture existed).
> 
> We tried this example on our project and it doesn't reproduce the issue unfortunately...
> 
> Any thoughts would be appreciated.

Sorry, this is my problem.This is a bug in `react-native-screens`.

**Comment by Slapbox** [comment]
> This is a bug in react-native-screens.

@NiuGuohui can you please share a link to that issue (if there is one) for future readers?

**Comment by NiuGuohui** [comment]
> > This is a bug in react-native-screens.
> 
> @NiuGuohui can you please share a link to that issue (if there is one) for future readers?

In my case, I want to achieve a similar effect to iOS's UITableView (such as horizontally laid out tabs). When I use this pattern nested in a native-stack, I encounter a similar issue. I think this might be a minor issue with the react-native-screens library. see [there](https://github.com/react-navigation/react-navigation/issues/2141)
Here is my code (in a native-stack screen):
```
<View style={{ flexDirection: 'row', height: '100%' }}>
  // this is real tabBar component
  <View style={{ width: 60, height: '100%' }} />
  <Tab.Navigator
    initialRouteName="One"
    sceneContainerStyle={{ flex: 1 }}
    screenOptions={{ headerShown: false }}
   // mock a empty tabBar
    tabBar={props => null}
  >
    <Tab.Screen name="One" component={One} />
  </Tab.Navigator>
</View>
```

**Comment by israelidanny** [comment]
I am using `expo-router` which I don't think is using `react-native-screens` and getting the exact same behavior. So it seems like a bug that transcends just one library?

@necolas would be very nice if someone from contributors acknowledges the existence of this, because it makes the app I'm working on have quite degraded UX. You've had a few reports here already?

**Comment by vrgimael** [comment]
We are experiencing this problem in some of our other apps too, this time with no map component mounted whatsoever (disproving my theory above). Closing and reopening the app mysteriously fixes the problem.

https://github.com/facebook/react-native/assets/14969223/8e829428-30a7-4311-9738-1cd1fb0e3457

I'm surprised there aren't many more people complaining about this issue, it certainly happens in many other cases. I can only assume people reopen the app without thinking about it much.

Thoughts?

**Comment by mtjkln** [comment]
Hello, Guys,

using React-Native 0.71.8 

When using new architecture the touchable works sometimes and sometimes it wont. The moment I build using old architecture it works perfectly. 

Please help us @necolas

**Comment by the-skins-app** [comment]
The issue is triggered when `useNativeDriver` is set to `true` as per this issue: https://github.com/facebook/react-native/issues/36504

`onPress` works until a translation animation -- with `useNativeDriver: true` -- occurs. Thereafter `onPress` never works again anywhere in the app. 

- android
- 71.10
- both debug + release modes

**Comment by israelidanny** [comment]
@the-skins-app I guess there are a multitude of bugs because I get mine on iOS (not android only) ...

**Comment by jugshaurya** [comment]
> The issue is triggered when `useNativeDriver` is set to `true` as per this issue: https://github.com/facebook/react-native/issues/36504
> 
> `onPress` works until a translation animation -- with `useNativeDriver: true` -- occurs. Thereafter `onPress` never works again anywhere in the app. 
> 
> - android
> - 71.10
> - both debug + release modes


I found a temp/possible solution for it, after researching for more than 2 weeks. 

Use `onPressIn` instead of `onPress` !

Let me know if it also works for you. Also to increase tappable area to make button more responsive, you can use `hitslop` prop.

**Comment by Crhaj** [comment]
> I tried to help by pointing out that a reproducible example will help the maintainers (not me) more easily look for the root cause of the problem.

@necolas we understand that reproducible example would help tremendously but it seems like no one **can** provide that reliably. This issue seems to be completely random @vrgimael even mentioned it is happening across multiple different versions of RN. For us it is only happening in one of our apps where, unfortunately, it is big problem as it needs to be constantly running and communicating with other apps. 

It really needs some attention, this is making all applications that need quick & reliable interactions and cannot just be restarted virtually useless.

@vrgimael have you possibly also tried swapping out Touchable components for Pressable? Does the issue persist?

**Comment by horne3754sg** [comment]
Hi all, I have been encountering this issue as well. 

I noticed that when I disable animations for react navigation, it seems to stop happening. When I enabled animations on navigators, it appears that randomly touch events stop responding, as well as the back button. 

I cant reliablity reproduce the issue though, it just happens at random.

**Comment by horne3754sg** [comment]
@vrgimael are you able to try and switch the Touch components to the ones from `react-native-gesture-handler`. You may need to adjust some styling or set the container styling. But its 99% the same as the `react-native` touch components.

**Comment by vrgimael** [comment]
Hi @horne3754sg, we have already tried it and the same problem still happens. We were unable to tell whether it makes any different, but it doesn't seem like it

**Comment by horne3754sg** [comment]
@vrgimael I have been running into the same issues as well. I have not tried automating some tests.

I have switched all my Touchables to Pressables (which is recommended). But I am still hitting the same issue where they become unresponsive, but if I change the `onPress` prop to `onPressIn`, it works. The problem though, is that then causes other issues like in scroll views.

When I look at the log, I can see the different events being logged.

3 taps and the first 2 only inPressIn/onPressOut fire and on the 3rd tap, all 3 onPress events fire. 

The problem is I cannot see what is preventing onPress from firing the first 2 taps.

![image](https://github.com/facebook/react-native/assets/2529445/10be0c35-52b2-463e-a785-cc11995d274c)

**Comment by vrgimael** [comment]
Indeed, check point 6 of my OP:

_6. When the issue happens, the onPressIn and onPressOut events are fired almost simultaneously with only a few milliseconds difference but onPress is not fired - kinda like what happens when you're scrolling or actually interacting with another gesture handler. I tried to make a fallback mechanism that uses this behaviour to our advantage but it's outcome is not very consistent for a few different reasons._

I tried a solution using this and the x and y velocity to determine whether the user is tapping or trying to scroll, and although this seem to be the best workaround so far, it still doesn't work every time.

It just seems like some other touchable area is intercepting the touch event and preventing it from firing the actual "onPress" trigger of the desired target.

**Comment by vrgimael** [comment]
> The issue is triggered when `useNativeDriver` is set to `true` as per this issue: #36504
> 
> `onPress` works until a translation animation -- with `useNativeDriver: true` -- occurs. Thereafter `onPress` never works again anywhere in the app.
> 
> * android
> * 71.10
> * both debug + release modes

Regarding this - we observed the problem in both Android and iOS, so it might not actually be the same problem, although I'll do some more tests with animations as something related seems plausible.

**Comment by Crhaj** [comment]
> > The issue is triggered when `useNativeDriver` is set to `true` as per this issue: #36504
> > `onPress` works until a translation animation -- with `useNativeDriver: true` -- occurs. Thereafter `onPress` never works again anywhere in the app.
> > 
> > * android
> > * 71.10
> > * both debug + release modes
> 
> Regarding this - we observed the problem in both Android and iOS, so it might not actually be the same problem, although I'll do some more tests with animations as something related seems plausible.

There is a note in the linked issue that the bug occurs on android when the new architecture is enabled but at least in our case, it is happening also on old architecture so I'm not sure it is the same problem.

**Comment by FluffyDiscord** [comment]
Also experiencing this issue on Android new arch enabled. For now my workaround for simulating `onPress` is using onTouchStart/end and move

**Comment by meabed** [comment]
Having the same issue exactly, android touchable stop responding randomly, also happens when there is an sync task or api call or exception in the background. 

It never happens on development ( only on build / dev / prod )

**Comment by meabed** [comment]
I have a case where i could reproduce this issue, and I just tested with `react-native 0.72.1` and the issue seems fixed. I will keep watching for it and update the issue incase if it happen again.

**Comment by dunapanta** [comment]
In my case, I've noticed an error occurring on the Onboarding screens with animations specifically on iOS. I will try using `onPressIn` to see if it resolves the issue. However, I will also stay updated on any new updates related to this problem.

**Comment by Crhaj** [comment]
> I have a case where i could reproduce this issue, and I just tested with `react-native 0.72.1` and the issue seems fixed. I will keep watching for it and update the issue incase if it happen again.

@meabed would it be possible yo share the code where you were able to reproduce this issue consistently?

**Comment by shawarmaz** [comment]
seeing this issue as well

**Comment by techfortified** [comment]
HOW TO FIX REACT NATIVE NOT RESPONDING TO CLICK EVENTS ON REAL DEVICES

Are you still facing this issue? I faced this same issue and it was so painful having a properly working app on emulators but buttons not responding to clicks on real devices.

Here are possible solutions that worked for me.

First of all install react-native-gesture-handler

`npm install react-native-gesture-handler`

import 'react-native-gesture-handler' at the top of your root app in *App.js* or  *index.js*

`import 'react-native-gesture-handler'`

In `App.js` wrap your App  with GestureHandlerRootView

`import { GestureHandlerRootView } from 'react-native-gesture-handler'`

`<GestureHandlerRootView style={{flex: 1}}>
     <App/>
</GestureHandlerRootView>`

Note: I used [react native paper](https://reactnativepaper.com/) you can follow the docs to install 

*Optionally, you can install the library or use any library.*


*Example ~ Works quite well on emulators but sometimes don't respond to clicks on real devices. With multiple press, it responds but we don't want that. We want our app to respond once pressed. This is really awful most especially in a ScrollView or FlatList*
```
import { Button }  from  'react-native-paper'
import {Alert} from 'react-native'

  const HomeScreen = () => {
      return (<Button onPress={() => Alert.alert("Hello world")>Press me</Button>)
  }
  export default HomeScreen
  
```
To fix the above code to respond fully well on real devices both android and iOS,  particularly android devices. This solutions work in ScrollView or FlatList

*SOLUTION 1*

1. Wrap all your components with gestureHandlerRootHOC
2. change `onPress` event handler to `onPressIn` event handler
It's crazy but it works

```
import { gestureHandlerRootHOC }  from  'react-native-gesture-handler'
import {Button} from 'react-native-paper'
import {Alert} from 'react-native'


  const HomeScreen = () => {
      return (<Button onPressIn={() => Alert.alert("Hello world") title="Press me" />)
  }
  export default  gestureHandlerRootHOC(HomeScreen)

  //this will work perfectly on both emulators and real devices

```
*SOLUTION 2*
1. Wrap all your components with gestureHandlerRootHOC
2. Wrap the *Button* component with any of the Touchable event handler.

*You can use TouchableOpacity but if you don't want the ripple effect, use TouchableWithoutFeedback.
Remember it's not just buttons but any component with onPress event handler can be wrapped with of TouchableWithoutFeedback or TouchableOpacity*

```
import React, {Fragment} from 'react'
import {TouchableOpacity, TouchableWithoutFeedback, gestureHandlerRootHOC} from 'react-native-gesture-handler'; 
import { Button }  from  'react-native-paper' 


  const HomeScreen = () => {
      return (<Fragment>
        
         <TouchableOpacity onPress={() => Alert.alert("Hello world")>
                    <Button >Press me </Button>
        </TouchableOpacity>
        
        <TouchableWithoutFeedback onPress={() => Alert.alert("Hello world")>
                    <Button >Press me </Button>
        </TouchableWithoutFeedback>

      </Fragment>)
  }
  export default gestureHandlerRootHOC(HomeScreen)
```
*Library versions*
```
"react-native": "^0.72.3"
"react-native-paper": "^5.10.1"
"react-native-gesture-handler": "^2.12.1",
```

These are the solutions that worked for me. My device was Android 10 API 29, development system MacBook Pro M2.

You can experiment more and try other possible solutions too. Happy coding :)

**Comment by Crhaj** [comment]
> I have a case where i could reproduce this issue, and I just tested with `react-native 0.72.1` and the issue seems fixed. I will keep watching for it and update the issue incase if it happen again.

@meabed kind reminder, would it be possible to share the code where you were able to reproduce this issue consistently? It would be greatly appreciated!

**Comment by mars-peng-lb** [comment]
I have the same problem. Disabling all react-navigation animations can solve the problem, but it is still a problem because we need animations.
If you don't need animations, you can just do:
`screenOptions={{
        animationEnabled: false
      }}`

**Comment by sneh2608** [comment]
> I have the same problem. Disabling all react-navigation animations can solve the problem, but it is still a problem because we need animations. If you don't need animations, you can just do it in your navigator: `screenOptions={{ animationEnabled: false }}`

@satya164 Can you please look into this.We have to disable all animations to get this to work properly.

**Comment by dibyopra** [comment]
I can provide a case:

https://github.com/facebook/react-native/assets/50160911/035e9353-ecf6-4342-80a0-fd5e7764a5c9

<img width="626" alt="image" src="https://github.com/facebook/react-native/assets/50160911/b960a9e0-b034-47d6-8d7e-29d6d1b614b9">

in my case I have 3 button component
when the last button component is not responding when touched but the others is working perfectly
i tried comment the last button component and the result is only first button component is working

**Comment by IsuruSathyajith** [comment]
> I have the same problem. Disabling all react-navigation animations can solve the problem, but it is still a problem because we need animations. If you don't need animations, you can just do it in your navigator: `screenOptions={{ animationEnabled: false }}`

This workaround supported to resolve this issue for me. However, it comes with the expense of animations on stack screen changes. 
Is there any ongoing discussion on React Navigation V6 to solve this anytime soon? (I couldn’t find)
For me; This issue existed only on devices running Android 13. For Example, I had the issue on a Pixel 6A & a Samsung S20 both of which were with Android 13.

**Comment by sneh2608** [comment]
Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler.
This works because Touchables in react-native-gesture-handler do not use JS Responder system.
[Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)

**Comment by IsuruSathyajith** [comment]
> Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler. This works because Touchables in react-native-gesture-handler do not use JS Responder system. [Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)

Does it make the keyboard touch events work as well? Currently keyboard doesn’t get poped up when clicked on a input field too.

**Comment by sneh2608** [comment]
> > Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler. This works because Touchables in react-native-gesture-handler do not use JS Responder system. [Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)
> 
> Does it make the keyboard touch events work as well? Currently keyboard doesn’t get poped up when clicked on a input field too.

Yes...for that you will have to use TextInput also from react-native-gesture-handler

**Comment by horne3754sg** [comment]
> I have the same problem. Disabling all react-navigation animations can solve the problem, but it is still a problem because we need animations. If you don't need animations, you can just do it in your navigator: `screenOptions={{ animationEnabled: false }}`

This is what I have had to do for the time being on Android. Hopefully a bug fix can be provided in the near future.

**Comment by the-skins-app** [comment]
My company is willing to put a bounty of $1,000 on fixing this and getting it merged. If anyone else would like to add to the bounty, please chime in.

The reason this needs to be top priority is because concurrent react has been out for a long time, yet -- in practice -- not in the react-native world, as this issue makes Android useless, given all workarounds are unacceptable for pro apps. 

For additional information, we currently have an app in both app stores, and I implemented an `onPressIn` hack + a timeout after scrolling so events aren't triggered while scrolling, and it's not satisfactory. The `onPress` prop of Pressable really needs to operate correctly for the new architecture to be ready for prime time.

**Comment by israelidanny** [comment]
I’m shocked that the maintainers are ignoring this.

**Comment by Christofon** [comment]
> Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler. This works because Touchables in react-native-gesture-handler do not use JS Responder system. [Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)

This works for me as well and I will keep it like that until fixed. 

One interesting thing I noticed was, when only having the GestureHandlerRootView and a Button in the App component, everything works flawlessly. As soon as react-navigation get's added, it stops working again.

**Comment by IsuruSathyajith** [comment]
> > Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler. This works because Touchables in react-native-gesture-handler do not use JS Responder system. [Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)
> 
> This works for me as well and I will keep it like that until fixed.
> 
> One interesting thing I noticed was when only having the GestureHandlerRootView and a Button in the App component, everything works flawlessly. As soon as react-navigation get's added, it stops working again.

I tried creating a release build for Android without doing any workarounds we discussed and on the release build, all touch events worked fine, even on the same device I was trying before. Can you guys also try a release build and see what happens?

**Comment by Christofon** [comment]
> > > Another workaround to solve this issue is to wrap your entry file in GestureHandlerRootView from react-native-gesture-handler and use Touchables from react-native-gesture-handler. This works because Touchables in react-native-gesture-handler do not use JS Responder system. [Docs](https://docs.swmansion.com/react-native-gesture-handler/docs/installation/#js)
> > 
> > 
> > This works for me as well and I will keep it like that until fixed.
> > One interesting thing I noticed was when only having the GestureHandlerRootView and a Button in the App component, everything works flawlessly. As soon as react-navigation get's added, it stops working again.
> 
> I tried creating a release build for Android without doing any workarounds we discussed and on the release build, all touch events worked fine, even on the same device I was trying before. Can you guys also try a release build and see what happens?

Same behaviour in release build in my case. Already tried that. This also seems related to the Android version, does not happen on all our test devices.

**Comment by IsuruSathyajith** [comment]
> Same behaviour in release build in my case. Already tried that. This also seems related to the Android version, does not happen on all our test devices.

Yes, I mostly had it on Android 13 devices; so I created this issue for the React Navigation library back then -> https://github.com/react-navigation/react-navigation/issues/11619

**Comment by souza-dev** [comment]
> My company is willing to put a bounty of $1,000 on fixing this and getting it merged. If anyone else would like to add to the bounty, please chime in.
> 
> The reason this needs to be top priority is because concurrent react has been out for a long time, yet -- in practice -- not in the react-native world, as this issue makes Android useless, given all workarounds are unacceptable for pro apps.
> 
> For additional information, we currently have an app in both app stores, and I implemented an `onPressIn` hack + a timeout after scrolling so events aren't triggered while scrolling, and it's not satisfactory. The `onPress` prop of Pressable really needs to operate correctly for the new architecture to be ready for prime time.

All I had to do was disable the fabric in gradle.

**Comment by rupeshkk-iprogrammer** [comment]
any update on the touchable components?

**Comment by Roshett** [comment]
I encountered with the same bug

**Comment by sagarjeu** [comment]
Any update @satya164

**Comment by javache** [comment]
We expect some of this to be improved by https://github.com/facebook/react-native/commit/03f70bf995379f08a77abcf96bb0e31ff75ca8c3 which will ship in 0.73.

**Comment by Crhaj** [comment]
Thank you @javache, would it be possible to consider this to be also released as a patch for 0.72?

**Comment by Siti-Shaufi** [comment]
> Thank you @javache, would it be possible to consider this to be also released as a patch for 0.72?

Currently on react-native 0.72.6 and react 18.2.0 with expo. All with latest dependencies. Still same issue. :(

**Comment by ravindraguptacapgemini** [comment]
I am also facing the same issue, specially on Android devices. Clicks do not work randomly but the scroll works fine at the same moment. Not sure why React Native/Meta developers are not fixing it, as this is very prominent issue.
@cortinico

**Comment by ravindraguptacapgemini** [comment]
> We expect some of this to be improved by [03f70bf](https://github.com/facebook/react-native/commit/03f70bf995379f08a77abcf96bb0e31ff75ca8c3) which will ship in 0.73.

@javache I tried to patch this, but it did not work.

**Comment by ryanstout** [comment]
Same issue, our app is only for Android. Works in the emulator. Tried the latest RN and a few before. Thanks

**Comment by aatiqJalanSWE** [comment]
Not working mostly for below android 11.

**Comment by the-skins-app** [comment]
Is this working finally with RN 73? This is so important that Android is useless without this. Very sad. 

It seems that at this point this would be a low enough hanging fruit providing enormous value that it is finally worth prioritizing.

**Comment by faceyspacey** [comment]
Looking forward to being able to use Android with the new architecture finally. Waiting on this too.

**Comment by Wahsner** [comment]
Still appear to be having this issue in 0.73.X on Android where the onPress isnt being triggered on, in this instance, the RN <Button /> component.

If I disable newArchEnabled the button handles the onPress everytime.

Is anyone else still having this issue?

**Comment by mtjkln** [comment]
@Wahsner 

Yeah, exact case when it stops working when you turn on new arch.

Its very frustrating, that it still exists.

**Comment by christophliftoff** [comment]
Also experiencing this problem. It seems to be related to react-navigation, as all the reported instances from our users occur when the user attempts to navigate to a new screen. For example, user taps on the "back" button, no navigation occurs, but the back button's opacity changes, indicating that the button is partly responding to the touch.

As other's have reported on this GH issue, other components continue to work fine (such as scrollviews, etc). `TouchableOpacity` components stop working until the app is force closed and re-opened.

**Comment by cortinico** [comment]
Do we have a repro of any sort?

**Comment by selvakumar-aspire** [comment]
I had a similar issue, on Android 14

I replaced
import { createStackNavigator } from '@react-navigation/stack';
with
import { createNativeStackNavigator } from '@react-navigation/native-stack';

Now it works fine.

**Comment by cipolleschi** [comment]
Hi everyone, we might have landed a fix in main for this and we might be able to verify the fix using a nightly version.

Has any of you a repro that we can use to see if it is fixed? Pinging some of the recently active people: @christophliftoff, @mtjkln, @Wahsner, @faceyspacey, @the-skins-app?

**Comment by joemun** [comment]
> Hi everyone, we might have landed a fix in main for this and we might be able to verify the fix using a nightly version.

@cipolleschi Mind sharing the relevant commits? We can try to test as well, thanks!

**Comment by Siti-Shaufi** [comment]
> Hi everyone, we might have landed a fix in main for this and we might be able to verify the fix using a nightly version.
> 
> Has any of you a repro that we can use to see if it is fixed? Pinging some of the recently active people: @christophliftoff, @mtjkln, @Wahsner, @faceyspacey, @the-skins-app?

Can we know the solution, currently in 72.6 with the newArchitecture disabled and have react-navigation also have the same issue.

**Comment by mtjkln** [comment]
@Siti-Shaufi 

The issue happens when new architecture is enabled. Apart from this I have never come across this issue.

**Comment by mtjkln** [comment]
@cipolleschi 

The issue happens when new architecture is enabled. Apart from this I have never come across this issue.

**Comment by Crhaj** [comment]
> @cipolleschi
> 
> The issue happens when new architecture is enabled. Apart from this I have never come across this issue.

We (and other people here) have experienced the issue on the old architecture as well, so I don't think the architecture is the determining factor.

**Comment by cipolleschi** [comment]
@joemun sure! These are the relevant commits:
- Jan 4: https://github.com/facebook/react-native/commit/e4708d661bf3d01ec857635f04a4aabf9d954a5e
- Jan 2: https://github.com/facebook/react-native/commit/54166342f01fc74236ba6146a4c0f604189017e4

**Comment by hollanderbart** [comment]
In which next version will these commits be included?

**Comment by oliverphardman** [comment]
Following as I am also experiencing this, even on Android 5, on old architecture.

**Comment by cipolleschi** [comment]
@hollanderbart for the time being, we are considering these for 0.74.

If this is something that is crippling your 0.73 app, please add them as a pick request in the [discussion](https://github.com/reactwg/react-native-releases/discussions/98)

**Comment by Crhaj** [comment]
Requested in the discussion thread. Please upvote if you want it cherry picked to 0.73

**Comment by hollanderbart** [comment]
Did anyone tested if the relevant commit(s) actual resolved the issue? @joemun @Siti-Shaufi

**Comment by Bowlerr** [comment]
Having the same issue with react-native-gesture-handler on Android, are there any solutions to this?

**Comment by cipolleschi** [comment]
@Crhaj, @joemun @Siti-Shaufi @Bowlerr can you try if the commits I [linked above](https://github.com/facebook/react-native/issues/36710?fbclid=IwAR3oWTsJSeQgTewOgIz9B0Y-cYW92dCkIcN1vc1x6CtJndJt_WYRgc0DBmo#issuecomment-1882841453) fixes the issues for you? 

Or can you provide a [reproducer](https://github.com/react-native-community/reproducer-react-native) so I can try that out myself? (Pro tip: you can use 0.73 to repro the failure and then use a nightly to verify that it is solved)

Thank you so much! 🙏

**Comment by Crhaj** [comment]
> @Crhaj, @joemun @Siti-Shaufi @Bowlerr can you try if the commits I [linked above](https://github.com/facebook/react-native/issues/36710?fbclid=IwAR3oWTsJSeQgTewOgIz9B0Y-cYW92dCkIcN1vc1x6CtJndJt_WYRgc0DBmo#issuecomment-1882841453) fixes the issues for you?
> 
> Or can you provide a [reproducer](https://github.com/react-native-community/reproducer-react-native) so I can try that out myself? (Pro tip: you can use 0.73 to repro the failure and then use a nightly to verify that it is solved)
> 
> Thank you so much! 🙏

@cipolleschi unfortunately, in our case, we couldn't reliably reproduce the issue and it seemed to happen randomly.

**Comment by Bowlerr** [comment]
Same with @Crhaj, We couldn't reproduce the issue reliably.

**Comment by joemun** [comment]
> can you try if the commits I [linked above](https://github.com/facebook/react-native/issues/36710?fbclid=IwAR3oWTsJSeQgTewOgIz9B0Y-cYW92dCkIcN1vc1x6CtJndJt_WYRgc0DBmo#issuecomment-1882841453) fixes the issues for you?

@cipolleschi Sorry for the delay. I'm sad to say that those 2 commits (cherry-picked to our 72.3 branch) don't fix the issue for us.

**Comment by the-skins-app** [comment]
The issue only occurs when `useNativeDriver` is set:

https://github.com/facebook/react-native/issues/36504#issuecomment-1586137079

So if various automated tests are neglecting to set that to `true`, then you won't see the issue. @sammy-SC at Facebook seems to have figured that out as he just re-opened the above related issue.

**Comment by cipolleschi** [comment]
@joemun, thanks for letting us know.

We have this investigation marked as high priority, so we hopefully would be able to make progress on this as fast as we can.

To help us out, we would greatly appreciate a reproducer starting from [this repo](https://github.com/react-native-community/reproducer-react-native). I understand that the issue does not reproduce consistently, but if you can set up something and tell us how and how much we need to interact with the repro to get the issue happening, this will help us a lot.

Thank you all for the cooperation! 🙏

**Comment by rupeshkk-iprogrammer** [comment]
The issue happens when new architecture is enabled and also in  Reac-Native 73 new arch = true. Apart from this I have never come across this issue.

**Comment by kchh90** [comment]
I can confirm we're experiencing this issue as well on react-native 73.2 when new architecture is enabled for Android. The issue we face is with onPress only working intermittently on physical android devices. The issue does not occur on android emulator devices.

I have been able to consistently reproduce this issue with the following [repro](https://github.com/kchh90/repro-36710).

**Comment by cortinico** [comment]
> @cipolleschi I have been able to consistently reproduce this issue with the following [repro](https://github.com/kchh90/repro-36710) by clicking the "CLICK ME" button multiple times and not seeing each click generate the expected timestamp output in the console.

@kchh90 thanks for the reproducer. However, it seems to work fine on my end. See the attached video:


https://github.com/facebook/react-native/assets/3001957/3af1c69e-c14b-4a6d-a9a0-6a47cb15cf20

**Comment by Christofon** [comment]
> > @cipolleschi I have been able to consistently reproduce this issue with the following [repro](https://github.com/kchh90/repro-36710) by clicking the "CLICK ME" button multiple times and not seeing each click generate the expected timestamp output in the console.
> 
> @kchh90 thanks for the reproducer. However, it seems to work fine on my end. See the attached video:
> 
>  Screen.Recording.2024-01-26.at.09.54.20.mov

This is only happening on physical devices, at least for me.

**Comment by kchh90** [comment]
> > @cipolleschi I have been able to consistently reproduce this issue with the following [repro](https://github.com/kchh90/repro-36710) by clicking the "CLICK ME" button multiple times and not seeing each click generate the expected timestamp output in the console.
> 
> @kchh90 thanks for the reproducer. However, it seems to work fine on my end. See the attached video:
> 
>  Screen.Recording.2024-01-26.at.09.54.20.mov

Yes sorry, I should have been more clear that the issue is only reproducible on physical android devices but not on android emulators.

**Comment by kchh90** [comment]
Here's a short recording of the issue occurring on a physical android device. I added an alert to the button's onPress behaviour to better showcase the issue directly on the device:
https://github.com/facebook/react-native/assets/1700361/21a28388-df28-492b-bfc0-9d3150866250

I'm running it on Android 14 Pixel 7 in this example, but have tried it on a few other physical android devices and see the same issue, so doesn't seem to be device specific.

**Comment by cortinico** [comment]
Hey @kchh90 thanks for your reproducer, after furhter investigation, we were able to reproduce (I'd like to stress once more how having a simple/small reproducer was crucial here! Thank you again).

1. The issue happens on either emulator or physical device. The issue happens if you touch, move the finger, and release it.
2. The issue seems to be related to `react-native-screens`. It won't happen on a vanilla native app (cc @WoLewicki)

We'll follow up with next steps once we have a solution for this

**Comment by kchh90** [comment]
@cortinico Awesome, thank you very much for the effort in looking into this issue!

Through further testing, I've discovered that disabling the header for a given screen also appears to workaround this issue.

For example, by adding the following options to the screen in the repro as such:
```
<Stack.Screen
  name="Test"
  component={TestScreen}
  options={{
    headerShown: false,
  }}
/>
```

**Comment by etodanik** [comment]
I get this issue on expo-router without using screens. Not sure the current understanding of the bug is accurate if the assumption is that screens is a culprit.

**Comment by cortinico** [comment]
> I get this issue on expo-router without using screens. Not sure the current understanding of the bug is accurate if the assumption is that screens is a culprit. @cortinico

So the issue in the reproducer happens because of usages of `react-native-screens`. We'll have to patch the library to fix it. `expo-router` might also be affected by a similar problem, we haven't investigated it yet.

**Comment by Wahsner** [comment]
> I get this issue on expo-router without using screens. Not sure the current understanding of the bug is accurate if the assumption is that screens is a culprit. @cortinico
> 
> I'll try to extract a minimal repro but the issue is very intermittent and elusive.

expo-router is built on react-navigation according to their docs. https://docs.expo.dev/router/introduction/

**Comment by WoLewicki** [comment]
@etodanik by saying that you did not use screens, you mean you explicitly set this prop: https://reactnavigation.org/docs/stack-navigator/#detachinactivescreens to `false` ?

 Also, @cortinico it kinda looks the same issue we reported quite a long time ago: https://discord.com/channels/514829729862516747/702994331711045732/1130887943867875488. It was then fixed (probably) by this PR: https://github.com/facebook/react-native/pull/37779. But we've just recently started seeing the issue again, so we'll have to investigate it once more.

**Comment by cortinico** [comment]
> Also, @cortinico it kinda looks the same issue we reported quite a long time ago: [discord.com/channels/514829729862516747/702994331711045732/1130887943867875488](https://discord.com/channels/514829729862516747/702994331711045732/1130887943867875488). It was then fixed (probably) by this PR: #37779. But we've just recently started seeing the issue again, so we'll have to investigate it once more.

@WoLewicki that's correct. The issue is sort of similar

**Comment by WoLewicki** [comment]
@etodanik @kchh90 @vrgimael can you check if applying https://github.com/software-mansion/react-native-screens/pull/2028 fixes your issue?

**Comment by kchh90** [comment]
@WoLewicki I can confirm applying [git@github.com:software-mansion/react-native-screens.git](https://github.com/software-mansion/react-native-screens/pull/2028) does indeed resolve this issue on the repro app, I appreciate the fast response in looking into this!

**Comment by efstathiosntonas** [comment]
@WoLewicki the fixes are for new arch only, not paper, , right?

**Comment by idrakimuhamad** [comment]
I'm facing similar (at least sounded similar), which cannot be replicated in a consistence manner, and the only consistency I can gather is it happened on Android almost all the time. We have almost no report that this happened on iOS, though we not ruling it out. It usually happened after playing the app for awhile, but there are cases also where it happened rather quickly. Same as others, the only way to get out of the froze is to kill and restart the app.

I can't give a re-producible repo obviously, nor I have recording, but when this happened, everything within a screen doesn't respond to any tap. Pressing buttons, text input, header back button, and physical button doesn't work. And if it suddenly happened during entering text in TextInput, I notice that text formatting doesn't work (I did some currency formatting when user is typing), as if the JS side just stop. The only thing that work is scrolling.

I'm thinking that this relates to performance of the app perhaps, so I did run system trace on my low end Android (Samsung A13) but I'm not familiar with reading the profile. I notice when the freezing happened, the mqt_js are not busy and nothing happened there. But render thread are definitely busy and not stopping some stuff and shown as "incomplete".

<img width="971" alt="Screenshot 2024-02-19 at 11 21 33 PM" src="https://github.com/facebook/react-native/assets/1400323/48555031-04ec-425e-99bb-53d57752ec9d">

I'm about to try to patch my RN using the touchable fixes, and test it again.

I'm on RN 0.71.16 and not using new arch.

**Comment by HadiModarres** [comment]
This is not an old arch problem, if something so critical like detecting presses on screen didn't work properly, React Native would have never reached the adoption and popularity that it did.

**Comment by meabed** [comment]
This definitely not old arch / new arch issue, it's animation issue, 
we disabled async calls and animations on buttons and navigation to get Android working.

**Comment by cortinico** [comment]
Hey @idrakimuhamad @HadiModarres @meabed 
Please do not highjack the conversation here.

We identified the issue as being related to react-native-screens on New Architecture. The fix is forthcoming and has been confirmed and working by @kchh90.
If you have other issues with touchable, please open a separate issue and provide a reproducer.

**Comment by Crhaj** [comment]
@cortinico the original author of this issue however said he was not using the new architecture and several others (myself included) confirmed it is happening on old architecture as well. I mean no offense but if anything, the conversation has been hijacked by new architecture.

**Comment by cortinico** [comment]
> the original author of this issue however said he was not using the new architecture and several others (myself included) confirmed it is happening on old architecture as well. I mean no offense but if anything, the conversation has been hijacked by new architecture.

Still to this day, no reproducer was provided to the problem you're referring to @Crhaj. The only meaningful reproducer which was effectively provided was the one from @kchh90 which was only on New Architecture. So we consider this issue related to that.

@Crhaj if you have a different issue, on Old/New Architecutre, **with a valid reproducer** please open a new issue and we will look into it. Commenting on this issue that you're also affected without providing a way to reproduce the issue, won't allow us to fix the problem at all.

**Comment by Crhaj** [comment]
@cortinico I understand. However, all I am saying is that the original opened issue is not answered and it was hijacked by (probably) different one. There is no reproducer because no one has been able to reproduce this reliable. This was not meant as a complaint that the issue is not solved but the fact is that this issue was opened in regards to something that is not solved (again probably due to the randomness of the occurence) and hence, to me, it makes sense it should not be closed or considered as such.

**Comment by cortinico** [comment]
> This was not meant as a complaint that the issue is not solved but the fact is that this issue was opened in regards to something that is not solved (again probably due to the randomness of the occurence) and hence, to me, it makes sense it should not be closed or considered as such.

We'll be closing this once https://github.com/software-mansion/react-native-screens/pull/2028 is merged and shipped in a new version of `react-native-screens`. If there is another issue which is not fixed by the PR I linked, you should open a new issue, and provide a reproducer.

**Comment by Prasiddha-n** [comment]
I am having this issue for both iOS and android. But the peculiar thing about this is that it is working as expected and touches are registering in the test environment (Expo Go, `npx expo run:ios` and `npx expo run:android`). But as soon as I create a build using `eas build -p ios/android -e "development" --non-interactive` and submit to appstoreconnect or create release for internal testing in android, then the app does not register the touches. Scroll horizontal and vertical are working as expected but it is not registering any kind of touch/press.

My project was on expo 48 and I just updated it to expo 50. I further updated other packages using `npx expo install  --check`.

@cortinico  I was just wondering, the fix from https://github.com/software-mansion/react-native-screens/pull/2028 will resolve that too? Please some insights on this will be very helpful.

**Comment by felix-lambert** [comment]
I'm also having this issue with expo-router and the new architecture. If i navigate to a new screen no touchable events are responding. Do you need to have a minimal reproduction of the problem ?

**Comment by cipolleschi** [comment]
@felix-lambert yes, please, but possibly without expo router. We need reproducers only using core, if possible. If the failing behavior happens after navigation only, is highly likely that the navigation library is the root cause of the issue. 
We worked with RNScreens maintainers and we already found an issue that they should have solved.

**Comment by felix-lambert** [comment]
@cipolleschi here is a minimal reproduction without expo-router. https://github.com/felix-lambert/touchablenotworking. Hope that will help

**Comment by satya164** [comment]
@felix-lambert is this different from https://github.com/software-mansion/react-native-screens/pull/2028 ?

**Comment by felix-lambert** [comment]
Hello @satya164. I `yarn add "software-mansion/react-native-screens#@wolewicki/proper-yoga-on-fabric"`and the onPress seems to work. However the navigation.navigate does not work anymore. If i do yarn ass `react-native-screens` the onPress does not work but the navigation do. Is that normal? Here is my mini repro > https://github.com/felix-lambert/touchablenotworking . Hope that is helping

**Comment by umarzaman** [comment]
I am having the same issue on Expo Go cant tap on any TextInput, touchable or buttons nothing: I am having this issue on iphone 12 mini other wise it works fine on the iphone 12, oneplus  or other android devices.

NOW THE WEIRD PART WHEN I SHAKE THE IPHONE 12 MINI EVERYTHING WORKS FINE . TOUCHABLE, TEXTINPUTS EVERYTHING
ANY SOLUTION FOR THIS PROBLEM.

**Comment by hollanderbart** [comment]
I appreciate all these "hint" messages, but I believe it's more efficient to wait until the fix has been finished & merged.

https://github.com/facebook/react-native/issues/36710#issuecomment-1954411870

**Comment by felix-lambert** [comment]
The problem is still there even after the merge (https://github.com/software-mansion/react-native-screens/pull/2028). Is there an alternative solution? Thanks for the help :)

**Comment by efstathiosntonas** [comment]
@felix-lambert are you testing against Fabric or Paper? The PR is targeting Fabric

**Comment by felix-lambert** [comment]
@efstathiosntonas i'm testing against Fabric in my expo project with 

` "plugins": [
      [
        "expo-build-properties",
        {
          "ios": {
            "newArchEnabled": true,
            "flipper": false
          },
          "android": {
            "newArchEnabled": true
          }
        }
      ],`

If I disable the new architecture it works. I'm using "react-native-screens": "~3.29.0" and "react-native": "0.73.6". I think @umarzaman and @aatiqJalanSWE is having the same problem

**Comment by cortinico** [comment]
We're closing this as the issue has been fixed on react-native-screen's end:
https://github.com/software-mansion/react-native-screens/pull/2028

If you're still facing similar issues, even after 0.74 + the react-native-screens patch, please open a separate issue and provide a reproducer.

**Comment by devym-37** [comment]
@cortinico 
We updated react-native-screens to the latest version, but the above issues still occur. This issue should be reopened. 
1. freezing on click while changing tabs via bottom tab, scrolling is possible, but events on subsequent touches are unresponsive.
2. freezing occurs when launching the app and then launching the app again in the background or sleep.

**Comment by devym-37** [comment]
@vrgimael Has this issue been resolved by any chance?

**Comment by cortinico** [comment]
> We updated react-native-screens to the latest version, but the above issues still occur. This issue should be reopened.

@devym-37 please open a new issue + provide:
* your react-native version
* the react-native-screens version
* a reproducer using the https://github.com/react-native-community/reproducer-react-native

**Comment by amanShukla-amber** [comment]
> > This was not meant as a complaint that the issue is not solved but the fact is that this issue was opened in regards to something that is not solved (again probably due to the randomness of the occurence) and hence, to me, it makes sense it should not be closed or considered as such.
> 
> We'll be closing this once [software-mansion/react-native-screens#2028](https://github.com/software-mansion/react-native-screens/pull/2028) is merged and shipped in a new version of `react-native-screens`. If there is another issue which is not fixed by the PR I linked, you should open a new issue, and provide a reproducer.

As the above fix for React native screens is pushed into version 3.30.1, Can't we directly upgrade the React-native-screens version above 3.30.1 in our app instead of upgrading React Native version ?

**Comment by WoLewicki** [comment]
@amanShukla-amber you can try and apply only this PR to the current version of `react-native-screens` you have by using e.g. [patch-package](https://www.npmjs.com/package/patch-package). From quickly looking at the code, it might just work on previous versions also. Be sure to apply https://github.com/software-mansion/react-native-screens/pull/2107 too then.

**Comment by spyshower** [comment]
@vrgimael thank you for the detailed post man. We too have been struggling for months. Any chance you guys found out the issue?

**Comment by emanu7478** [comment]
> @vrgimael thank you for the detailed post man. We too have been struggling for months. Any chance you guys found out the issue?

"Hi @spyshower, We've been dealing with a similar issue for quite some time. Luckily, after thorough investigation, we managed to resolve it by upgrading the react-native-screen package to its latest release, version 3.31.1. Hopefully, this solution works for you too!"

**Comment by peaktan** [comment]
Hello, we have encountered this problem, could you tell me the latest version of the repair, see a fix to repair a similar problem before: https://github.com/facebook/react-native/pull/45865, don't know whether to fix the problem @vrgimael

**Comment by sakibb7** [comment]
I'm using  "react-native-screens": "~4.1.0" but still facing the same issue. onPress works but i need to click twice. This is very annoying. onPress works perfectly on android emulator but not on physical device.

**Comment by cipolleschi** [comment]
@sakibb7 the latest version of Screens is 4.4.0. Are you using RN Gesture Handler as well? If not, can you try to use it and report if the problem persists?

**Comment by Atom1er** [comment]
This issue should be re-opened, this is driving us crazy, it is random, we tried using different type of touchable to no luck. Was happening on an old version, so we updated to the latest and it still happens.

**Comment by Bowlerr** [comment]
@Atom1er have you checked for any memory leaks? It has been over a year but that ended up being a part of the issue I was having back then. 

Touchables would stop being responsible after a while due to the memory leak

**Comment by RohovDmytro** [comment]
I am investigating very, very random stop on onPress for a long time. I had to invent the concept of hope to survive.

onPress => Taps => Stop working. Google Pixel 7 phone, no error, nothing sus in the long.

I am using BaseButton (`react-native-gesture-handler`) + Text component. Replacing with a basic react-native `Button` component solves the issue.

**Comment by RohovDmytro** [comment]
Do anybody anyhow anyway anychance has an estimation of a real-life production impact of this issue? Is it in the ranges of 1%, 10%, *?

**Comment by spyshower** [comment]
> Do anybody anyhow anyway anychance has an estimation of a real-life production impact of this issue? Is it in the ranges of 1%, 10%, *?

Im only replying because your comments made me laugh. If you ask in how many devices it happens, definitely <1%

**Comment by Siti-Shaufi** [comment]
Hi everyone,  

Our team hasn’t come across these issues since we decided to revamp the UI/UX for the entire app. We’re still looking into the root cause, but we suspect it might be related to memory leaks. What’s weird is that while it happens randomly, it always affects the same button on real devices but doesn’t show up on emulators. 

We also decide to just downgrade some of dependencies. 

```
 "react": "18.0.0",
"react-native": "0.69.12",
"react-native-screens": "^3.5.0",
```

**Comment by spyshower** [comment]
@Atom1er @Siti-Shaufi I am willing to help you out with this problem if you are interested. While we hadn't fixed the problem in the company I was working for at the moment, I do have some insights and potential solutions that might work.

**Comment by RohovDmytro** [comment]
I have some progress finding the cause.

At least SOME parts of the app were freezing due to `key` property. Basically:

1. Rendering an array of buttons each having `key={Math.random()}`
2. Quickly pressing works at first, then the app stops responding to pressed.

I was able to reliably reproduce errors by switching keys from random to constant values.

I have parts of the app where presses stops working where key value does not seem to be the issue. Will keep pushing (ha-ha).

**Comment by martinnov92** [comment]
Similar issue is hapenning to me, but only when phone goes to sleeps for some time, then I wake the phone up and suddenly nothing works, except scrolling and after some the the app crashes. Even AppState change is not called...

**Comment by spyshower** [comment]
> Similar issue is hapenning to me, but only when phone goes to sleeps for some time, then I wake the phone up and suddenly nothing works, except scrolling and after some the the app crashes. Even AppState change is not called...

Yep, exact same problem.

**Comment by martinnov92** [comment]
> > Similar issue is hapenning to me, but only when phone goes to sleeps for some time, then I wake the phone up and suddenly nothing works, except scrolling and after some the the app crashes. Even AppState change is not called...
> 
> Yep, exact same problem.

Did you find any solution?

**Comment by spyshower** [comment]
> > > Similar issue is hapenning to me, but only when phone goes to sleeps for some time, then I wake the phone up and suddenly nothing works, except scrolling and after some the the app crashes. Even AppState change is not called...
> > 
> > 
> > Yep, exact same problem.
> 
> Did you find any solution?

No man, read this thread. If someone wants to setup a chat/call so we can actually find out whats happening with this crazy issue instead of just reporting it, I will help.

**Comment by amanShukla-amber** [comment]
> > > This was not meant as a complaint that the issue is not solved but the fact is that this issue was opened in regards to something that is not solved (again probably due to the randomness of the occurence) and hence, to me, it makes sense it should not be closed or considered as such.
> > 
> > 
> > We'll be closing this once [software-mansion/react-native-screens#2028](https://github.com/software-mansion/react-native-screens/pull/2028) is merged and shipped in a new version of `react-native-screens`. If there is another issue which is not fixed by the PR I linked, you should open a new issue, and provide a reproducer.
> 
> [@cortinico](https://github.com/cortinico) As the above fix for React native screens is pushed into version 3.30.1, Can't we directly upgrade the React-native-screens version above 3.30.1 in our app instead of upgrading the React Native version? as we have upgraded the react-native-screens to the latest version with RN 0.71 but the issue is still there, do we have to upgrade React native version to 0.74+ to get this issue fixed?

We encountered a similar issue where touchables were not responding as expected. After extensive debugging, we discovered that the native thread was busy handling other tasks, which caused the press event callback to not propagate from the native side to the JS layer. As a result, the actions or code written in JS were not being executed.

To resolve this, we optimized our screen to reduce the workload on the native thread, and since then, we haven't observed this issue. This was tested on React Native version 0.71.

**Comment by kkafar** [comment]
I've been looking into the issue from `react-native-screens` perspective today. 

I've tried both reproductions provided above:

* https://github.com/felix-lambert/touchablenotworking
* https://github.com/kchh90/repro-36710

and my own playground. 

I've used Xiaomi Redmi Note 8 Pro with Android 11 (5 yo device) and failed to reproduce for ~20 min. In case anyone is able to reproduce it / record a video of the bug in your app / provide some other useful info - please do (together with versions of used packages).

**Comment by ChristopherGabba** [comment]
Hey everyone, I experienced this too in one of my screens.

Details:
- iOS
- react-native 0.76.7
- old architecture (paper)
- react-native-screens 4.4.0 (I tried the v4.9.1 too)
- @react-navigation/bottom-tabs 6.3.2
- @react-navigation/native 6.0.2
- @react-navigation/native-stack 6.0.2

We spent hours and hours debugging this. Basically we have a screen (think like snapchat stories) where I user could click a bunch of different videos (when pressing one, it opens up a "WatchingScreen" with a videoplayer in it) and close them one after the other. Seemingly at random, the user would close out of watching one of these and the enter "stories" panel was completely frozen. For some reason, the background scroll view could still be moved up and down but all the `Touchables` / `Pressables` on the page were completely locked up. It was a nasty bug. I also checked and I couldn't get `onPressIn()`, `onPressOut()`, or `onPress()` events to fire at all. We finally figured out the source for us:

```typescript
              <Stack.Screen
                name="Watching"
                component={Screens.WatchingScreen}
                options={{
                  // presentation: "containedTransparentModal", // CANNOT USE
                  // presentation: "transparentModal", // CANNOT USE
                }}
              />
```

I also think this could potentially be the same as the `<Modal />` component from `react-native` as well.

The conclusion we came to was that when calling `navigation.goBack()` on the "WatchingScreen" above and going back to this "stories page", somehow this `transparentModal` screen would not fully disappear? It's like it left some sort of invisible view barrier behind. This might not be the solution to some of your problems, but we are very confident this bug is "worked around" for now.

**Comment by shoxter** [comment]
Happening to me too... Have the same component FAB using TouchableOpacity -- works perfectly on a busy screen and on an empty screen touchable opacity will: sometimes work, sometimes display opacity change and not trigger on press, sometimes not do anything.

**Comment by kkafar** [comment]
Hey, I've recently landed 

* https://github.com/software-mansion/react-native-screens/pull/2819 

in `react-native-screens`, which potentially resolves problems some of you encounter. It addresses cases where you have `react-native-gesture-handler` and the pressables happen to lose focus. This does not solve cases, where touchables can't gain focus at all, however it's worth trying out. This fix is released with `react-native-screens@4.11.0-beta.0`.

I want to reiterate on my [previous comment](https://github.com/facebook/react-native/issues/36710#issuecomment-2616632609) if anyone affected is able to provide a reproduction - please do - it'll be great help.

**Comment by jorgetoh** [comment]
I'm experiencing this too with: "expo": "~52.0.43" and iPhone 16 Pro Max simulator.

**Comment by sergeymild** [comment]
for me touchable not working on android if i set presentation: 'formSheet', show screen as BottomSheet it causes some android devices skip touches

**Comment by mmarcinik** [comment]
for me this happened in a message list. After a certain number of pressable  elements are added to the list only a couple of the elements at the top are pressable. the rest are not.

**Comment by MandoMG** [comment]
> Am having issues after switching to RTL, any solutions?
> 
> RN : 0.76.9

I'm having this issue only with RTL as well. LTR works fine. Did you find any solutions? @Linuhusainnk 

using RN: 0.76.9 and react-native-screens 4.0.0.

**Comment by vaidyakhil** [comment]
Hey everyone, sharing this as a potential repro of this issue:
https://github.com/vaidyakhil/OnPressBugRepro

Hope this is helpful

- react-native-screens and any navigation library is **NOT being used** and it's a simple hello world App.
- **Have attached a screen recording for reference in the repo (`demo_screen_recording.mp4`)**
- Tested only on iOS simulator

---

I had just initialized a vanilla RN project with rn-cli when I faced this issue where sometimes a simple click listener was not getting called at all.

Eventually reached this issue. Based on the discussions above where it seems it was closed in favor of a `react-native-screens` issue.


https://github.com/user-attachments/assets/22a29ea5-80a8-4b8a-95c2-e488fb307589

**Comment by syahbes** [comment]
for me whats fixed this issue was to replace 'onPress' 
with 'onPressOut'

even with : 
```
<TouchableOpacityonPressOut={() =>  { conssole.log("clicked")}>
   <Text>Click me</Text>
</TouchableOpacity>
```

**Comment by onyedikachi23** [comment]
This also happens for Pressable. If only there was a workaround.

**Comment by MorganNavel** [comment]
I have the exact same problem as all of you. TouchableOpacity won't fire onPress  (and even onPressOut) : 

```ts
import { useThemeColor } from "@/hooks/useThemeColor";
import { TouchableOpacity, TouchableOpacityProps } from "react-native";

type ButtonProps = TouchableOpacityProps;
const Button = TouchableOpacity;


export function StyledButton({ style, ...props }: Readonly<ButtonProps>) {
  const backgroundColor = useThemeColor("primary");
  return <Button style={[style, { backgroundColor }]} {...props} />;
}
export { Button };


// Example : 
<StyledButton
      onPress={handleSubmit(onSubmit)}
      style={styles.btn}
      disabled={isLoading}
>
             
  <ThemedText
         type="defaultSemiBold"
         style={{
               color: onPrimary,
               textAlign: "center",
           }}
   >
       Click Me
   </ThemedText>
</StyledButton>
```

My concerned is that somehow when I only press on the right side (after the text) or the extreme left of my button, it will fire the event all the time, if I press in the middle or on the left part it doesn't work.

It happens both in debug and release.

I tested it on an Android 14
```JSON
{
  "dependencies": {
    "@expo/vector-icons": "^14.1.0",
    "@react-native-async-storage/async-storage": "^1.24.0",
    "@react-native-community/blur": "^4.4.1",
    "@react-native-firebase/app": "^22.4.0",
    "@react-native-firebase/messaging": "^22.4.0",
    "@react-navigation/bottom-tabs": "^7.3.10",
    "@react-navigation/elements": "^2.3.8",
    "@react-navigation/native": "^7.1.14",
    "expo": "~53.0.11",
    "expo-blur": "~14.1.5",
    "expo-checkbox": "^4.1.4",
    "expo-constants": "~17.1.6",
    "expo-font": "~13.3.1",
    "expo-haptics": "~14.1.4",
    "expo-image": "~2.3.0",
    "expo-linking": "~7.1.5",
    "expo-notifications": "^0.31.4",
    "expo-router": "~5.1.0",
    "expo-secure-store": "~14.2.3",
    "expo-splash-screen": "~0.30.9",
    "expo-status-bar": "~2.2.3",
    "expo-symbols": "~0.4.5",
    "expo-system-ui": "~5.0.8",
    "expo-web-browser": "~14.1.6",
    "i18next": "^25.3.1",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",
    "react-hook-form": "^7.58.1",
    "react-i18next": "^15.6.0",
    "react-native": "0.79.3",
    "react-native-collapsible": "^1.6.2",
    "react-native-device-info": "^14.0.4",
    "react-native-gesture-handler": "~2.24.0",
    "react-native-pager-view": "6.7.1",
    "react-native-reanimated": "~3.17.4",
    "react-native-safe-area-context": "5.4.0",
    "react-native-screens": "~4.11.1"
  }
}
```
EDIT: 
I managed to fix my problem. In my application, I was using a ScrollView and a Card component, and I was passing my button as a child. Initially, I realized that the issue was related (in my case) to elevation and z-index (for some reason, it would sometimes work and sometimes not, but it was consistent). Anyway, I had to explicitly add the z-index to my Card and my ScrollView.

**Comment by globalYK** [comment]
We also encountered this issue. Although it happens occasionally, the reproduction rate is quite high.

In our case, we wrapped a **TouchableOpacity** inside an **Animated.View**, and we were only able to receive onPressIn and onPressOut events — onPress was never triggered. We suspected that the problem lies in the JS side's handling of gesture events passed from native code.

To work around this, we switched to using **Pressable**, since it provides access to lower-level touch events. Then, we implemented our own custom onPress logic by mimicking how iOS handles tap gestures internally (based on distance and duration thresholds).

This approach successfully resolved the bug for us.

` 
const handlePress = () => {
    if (onPress && !disabled) {
      onPress()
    }
  }

  // iOS 使用 onPressIn 和 onPressOut 模拟 onPress，添加防滚动误触机制
  const isPressedRef = useRef(false)
  const pressStartTimeRef = useRef(0)
  const pressStartPositionRef = useRef({ x: 0, y: 0 })
  const movedBeyondSlopRef = useRef(false)

  const handleTouchStart = (event: any) => {
    logRNInfo('handleTouchStart')
    if (!disabled) {
      isPressedRef.current = true
      pressStartTimeRef.current = Date.now()

      // 记录按下时的位置 - 使用触摸事件的正确属性
      if (event && event.nativeEvent && event.nativeEvent.touches && event.nativeEvent.touches[0]) {
        const touch = event.nativeEvent?.touches?.[0]
        pressStartPositionRef.current = {
          x: touch.pageX,
          y: touch.pageY
        }
        logRNInfo('handleTouchStart pressStartPositionRef', pressStartPositionRef.current)
      }
    }
  }

  const handleTouchEnd = (event: any) => {
    logRNInfo('handleTouchEnd', event.nativeEvent?.touches?.[0])
    if (!disabled && isPressedRef.current) {
      const pressDuration = Date.now() - pressStartTimeRef.current
      const isShortPress = pressDuration <= kMaxTapDuration // 短按时间阈值
      logRNInfo('handleTouchEnd === pressDuration, isShortPress', pressDuration, isShortPress)
      
      // 只有短按且移动距离小时才触发点击
      if (isShortPress && !movedBeyondSlopRef.current) {
        handlePress()
        logRNInfo('handlePress')
      }
      
      isPressedRef.current = false
      movedBeyondSlopRef.current = false
    }
  }

  const handleTouchMove = (event: any) => {
    if (event && event.nativeEvent && event.nativeEvent.touches && event.nativeEvent.touches[0]) {
      logRNInfo('handleTouchMove', event.nativeEvent?.touches?.[0])
      const touch = event.nativeEvent?.touches?.[0]
      const {x, y} = touch
      const moveDistance = Math.sqrt(
        Math.pow(x - pressStartPositionRef.current.x, 2) +
        Math.pow(y - pressStartPositionRef.current.y, 2)
      )
      if (moveDistance > kMaximumTapMovement) {
        movedBeyondSlopRef.current = true
        logRNInfo('handleTouchMove movedBeyondSlopRef', movedBeyondSlopRef.current)
      }
    }
  }
`

**Comment by Dallas62** [comment]
Hello, just to add a small information, looks like the `locationY` is inconsistent on Android, while on iOS, looks correct. Found this on the `onPressOut` event.
Some times `locationY` is lower than the expected value. For exemple, the touch is outside of the element (height: 50) but the locationY is like 15. This seems to occur when another Pressable is close to the initial one. For example, list of button/pressable.
Some time `locationY` is equal to the `pageY`, when the touche occur outside of the screen.
Feels like there is some kind of reference shared across Pressable in the Android code.

**Comment by JamesMahy** [comment]
onPressOut seems to solve the issue for me on iOS on version `0.76` and the new architecture

**Comment by Dallas62** [comment]
@JamesMahy onPressOut is not a solution, it's an ugly workaround which also trigger when you scroll / move outside of the button.

**Comment by JamesMahy** [comment]
> [@JamesMahy](https://github.com/JamesMahy) onPressOut is not a solution, it's an ugly workaround which also trigger when you scroll / move outside of the button.

it's solved my issue for now mostly and I've added checks to detect things like Longpress in to it. 

But I agree - it needs a proper solution

**Comment by meabed** [comment]
For everyone in the thread this is what we identified the cause and what we changed to fix the issue:


We found the main issue happens is not only onPress it was also happening when the app opens or the we route go to screen.

So basically is not only on button, it was actually related errors / or code the through exceptions that happens in background.

For example we had axios api calls that happen on screen mount or button press or happening on setIntervals and whenever we reactive an error for example 500 from backend that cause unhandled exception this was randomly causing the app to stop responding 

We also have SQLite that sometimes SQLite connection or tables are not initialized yet before using the connection and it was also causing this background exception that are not handled.

So we went in all code based and we ensured 2 things:

- Any operation that may through error have a proper try cache handler properly.

- we fixed some types and used optional chaining properly in while we accessing data and methods as sometimes we access objects or methods and assumes they are always defined - but they are not - for example, SQLite.collection.find() need to become SQLite?.collection?.find() 


After that - the issue totally disappeared so my guess it's something in react native that triggers when this errors happens and cause the app to stop responding for some reason - but for sure it's not only the button press.


Hope you get the idea and this helps

**Comment by JamesMahy** [comment]
> For everyone in the thread this is what we identified the cause and what we changed to fix the issue:
> 
> We found the main issue happens is not only onPress it was also happening when the app opens or the we route go to screen.
> 
> So basically is not only on button, it was actually related errors / or code the through exceptions that happens in background.
> 
> For example we had axios api calls that happen on screen mount or button press or happening on setIntervals and whenever we reactive an error for example 500 from backend that cause unhandled exception this was randomly causing the app to stop responding
> 
> We also have SQLite that sometimes SQLite connection or tables are not initialized yet before using the connection and it was also causing this background exception that are not handled.
> 
> So we went in all code based and we ensured 2 things:
> 
> * Any operation that may through error have a proper try cache handler properly.
> * we fixed some types and used optional chaining properly in while we accessing data and methods as sometimes we access objects or methods and assumes they are always defined - but they are not - for example, SQLite.collection.find() need to become SQLite?.collection?.find()
> 
> After that - the issue totally disappeared so my guess it's something in react native that triggers when this errors happens and cause the app to stop responding for some reason - but for sure it's not only the button press.
> 
> Hope you get the idea and this helps

Thats isn't the same issue.
This bug relates to a pure Pressable / Touchable and a console.log on onPress.

It's intermittent and happens on only on the new architecture.

**Comment by Dallas62** [comment]
Here some details of the issue for Android: https://github.com/facebook/react-native/issues/53366

**Comment by spyshower** [comment]
> It's intermittent and happens on only on the new architecture.

Old and new.

**Comment by JamesMahy** [comment]
> > It's intermittent and happens on only on the new architecture.
> 
> Old and new.

I can't reproduce on old architecture.

**Comment by MaxiAschenbrenner** [comment]
Just wanted to ask if anyone has figured it out, or why this issue has been closed? 
We are encountering the same problem on Android using bare RN 81.4. 
OnPressIn and onPressOut are always firing, onPress sometimes doesnt. This happens both with the RN Pressable and the Pressable from gesture detector.
It only seems to happen in releases, not debug builds, which makes debugging this even more of a nightmare.

**Comment by meabed** [comment]
@MaxiAschenbrenner totally debugging is impossible for this it only happen on builds not on dev... 
We tried many different changes or different suggestions here but the issue was still happening.
the only thing that work for me was to ensure any thing happening like fetch / async operations in background or on navigation is wrapped in try catch with error handling - my guess in our case is something throwing an error in background like fetch error or access on undefined object and that was causing the navigation to freeze and buttons not to respond .. i am not sure how they are related but thats my observation. 

Try wrapping the code you have specially on the screen that is freezing in try cache with proper error handling and also some times make async then() so its not blocking for the navigation or screen components, hope this helps.

**Comment by MaxiAschenbrenner** [comment]
@meabed Thank you for the quick reply! 
I have read you comment above and and already tried this, it sadly didnt help.
One of the screens we can reproduce this on has three buttons, and only one exhibits this behaviour (not always the same button). OnPress also doesnt completely stop working, it just doesnt fire maybe 10% of the time, which imo further supports that it is not a general freeze.

```
            <ScrollView>
                {/* other children */}
                <Pressable onPress={() => {console.log('Button 1 pressed')}} /> 
                <Pressable onPress={() => {console.log('Button 2 pressed')}} />
                <Pressable onPress={() => {console.log('Button 3 pressed')}} />
            </ScrollView>
```

In the simplified scenario above, two buttons would always work, while the third will not log sometimes (onPressIn and out always work though). 

On another screen we we have a nested Pressable, where the child exhibits the same random behaviour. In this case, the onPress of the parent Pressable will log.

```
            <Pressable onPress={() => {console.log('parent')}}>
                {/* other children */}
                <Pressable onPress={() => {console.log('child')}} />
            </Pressable>
```

As i said in my comment above this happens both with the RN and the gesture handler Pressable.


As a workaround we are currently using a View wrapped in a GestureDetector with a tap gesture as a Pressable replacement on Android, but that is an ugly workaround for something that should work reliably out of the box.

**Comment by Qdafengzi** [comment]
Such a difficult problem to solve. The last time I upgraded to 0.79.2, because of this issue, I had to roll back to the old version.

This time I update RN to 0.82.1 problem still exist , if I change the TouchableOpacity from react-native-gesture-handler will be ok ,TouchableOpacity from react-native will be intermittent no response, and the response is also much slower.

SO COOL 🤪🤪🤪

**Comment by ermakov95** [comment]
Repeated in RN 0.81.6 + Wix-RNN 8.8.1

**Comment by returnMarcco** [comment]
We use react-navigation with react-native-screens.

We can reliably reproduce the issue in Android (iOS is fine) by:

1. Navigating to a screen
2. Rotating the phone (auto-rotate on) to landscape
3. Rotate phone back to portrait 
4. Navigate back to previous screen (nav.pop())
5. Pressables on all previous screens do not fire onPress when tapping, but do so if swiping left or right from within the affected Pressable 

Potential layout issue?

This has also occurred when triggering a Theme change from within the app, but this hasn't been reproducible.
